felt/src/lib.rs

```diff
@@ -40,16 +38,641 @@ pub trait FeltOps {
+    type Output = Felt;
+    fn mul(self, rhs: Self) -> Self::Output {
+        Self::Output {
+            value: &self.value * &rhs.value,
+        }
+    }
+}
+
+impl<'a> Mul<&'a Felt> for Felt {
+    type Output = Self;
+    fn mul(self, rhs: &Self) -> Self {
+        Self {
+            value: self.value * &rhs.value,
+        }
+    }
+}
+
+impl<'a> MulAssign<&'a Felt> for Felt {
+    fn mul_assign(&mut self, rhs: &Self) {
+        self.value *= &rhs.value;
+    }
+}
+
+impl Pow<u32> for Felt {
+    type Output = Self;
+    fn pow(self, rhs: u32) -> Self {
+        Self {
+            value: self.value.pow(rhs),
+        }
+    }
+}
+
+impl<'a> Pow<u32> for &'a Felt {
+    type Output = Felt;
+    fn pow(self, rhs: u32) -> Self::Output {
+        Self::Output {
+            value: (&self.value).pow(rhs),
+        }
+    }
+}
+
+impl Div for Felt {
+    type Output = Self;
+    fn div(self, rhs: Self) -> Self {
+        Self {
+            value: self.value / rhs.value,
+        }
+    }
+}
+
+impl<'a> Div for &'a Felt {
+    type Output = Felt;
+    fn div(self, rhs: Self) -> Self::Output {
+        Self::Output {
+            value: &self.value / &rhs.value,
+        }
+    }
+}
+
+impl<'a> Div<Felt> for &'a Felt {
+    type Output = Felt;
+    fn div(self, rhs: Self::Output) -> Self::Output {
+        Self::Output {
+            value: &self.value / rhs.value,
+        }
+    }
+}
+
+impl Rem for Felt {
+    type Output = Self;
+    fn rem(self, rhs: Self) -> Self {
+        Self {
+            value: self.value % rhs.value,
+        }
+    }
+}
+
+impl<'a> Rem<&'a Felt> for Felt {
+    type Output = Self;
+    fn rem(self, rhs: &Self) -> Self {
+        Self {
+            value: self.value % &rhs.value,
+        }
+    }
+}
+
+impl Zero for Felt {
+    fn zero() -> Self {
+        Self {
+            value: FeltBigInt::zero(),
+        }
+    }
+
+    fn is_zero(&self) -> bool {
+        self.value.is_zero()
+    }
+}
+
+impl One for Felt {
+    fn one() -> Self {
+        Self {
+            value: FeltBigInt::one(),
+        }
+    }
+
+    fn is_one(&self) -> bool {
+        self.value.is_one()
+    }
+}
+
+impl Bounded for Felt {
+    fn min_value() -> Self {
+        Self {
+            value: FeltBigInt::min_value(),
+        }
+    }
+
+    fn max_value() -> Self {
+        Self {
+            value: FeltBigInt::max_value(),
+        }
+    }
+}
+
+impl Num for Felt {
+    type FromStrRadixErr = ParseFeltError;
+    fn from_str_radix(string: &str, radix: u32) -> Result<Self, Self::FromStrRadixErr> {
+        Ok(Self {
+            value: FeltBigInt::from_str_radix(string, radix)?,
+        })
+    }
+}
+
+impl Integer for Felt {
+    fn div_floor(&self, rhs: &Self) -> Self {
+        Self {
+            value: self.value.div_floor(&rhs.value),
+        }
+    }
+
+    fn div_rem(&self, other: &Self) -> (Self, Self) {
+        let (div, rem) = self.value.div_rem(&other.value);
+        (Self { value: div }, Self { value: rem })
```

Comment:
Same, more complicated in fields.

Reply:
Oh, I think I missed something here. The `value` is already the field type, right? So the correct implementation is inside and this is just a forward to it, right? That would be true for the other cases as well, right?

Reply:
Resolved: https://github.com/lambdaclass/cairo-rs/pull/743/commits/dc4a90d5cb272a8880c1ab9397e3da2ee9c72ca2, https://github.com/lambdaclass/cairo-rs/pull/743/commits/0bf8a1dec86cb995d87b91f65924bb8ed174acee